src/main/java/co/cask/gcp/bigquery/BigQuerySink.java

```diff
@@ -114,17 +117,24 @@ public void prepareRun(BatchSinkContext context) throws Exception {
     configuration.set("fs.gs.project.id", projectId);
     configuration.set(BigQueryConfiguration.PROJECT_ID_KEY, projectId);
 
-    String temporaryGcsPath = String.format("gs://%s/hadoop/input/%s", config.bucket, uuid);
-    configuration.set("fs.gs.system.bucket", config.bucket);
+    String bucket = config.bucket;
+    if (config.bucket == null) {
+      bucket = uuid.toString();
+      // By default, this option is false, meaning the job can not delete the bucket. So enable it only when bucket name
+      // is not provided.
+      configuration.setBoolean("fs.gs.bucket.delete.enable", true);
```

Comment:
we still manually delete after the run, so what does this property do? Allow the bucket to be delete-able?

Reply:
Yes, we have to manually delete the bucket. This property makes it `delete-able`. By default, it can not be deleted.

Reply:
If we do not enable it, GoogleCloudStorageFileSystem logs warn message `Skipping deletion of buckets because enableBucketDelete is false` and does not delete the bucket.